source/server/admin/prometheus_stats.cc

```diff
@@ -285,6 +316,18 @@ std::string PrometheusStatsFormatter::formattedTags(const std::vector<Stats::Tag
   return absl::StrJoin(buf, ",");
 }
 
+absl::Status PrometheusStatsFormatter::validateParams(const StatsParams& params) {
+  switch (params.histogram_buckets_mode_) {
+  case Utility::HistogramBucketsMode::Summary:
+  case Utility::HistogramBucketsMode::Unset:
+  case Utility::HistogramBucketsMode::Cumulative:
+    return absl::OkStatus();
+  case Utility::HistogramBucketsMode::Detailed:
+  case Utility::HistogramBucketsMode::Disjoint:
+    return absl::InvalidArgumentError("unsupported prometheus histogram bucket mode");
+  }
```

Comment:
stupid gcc build gives this error:
```
2024-01-09T15:46:32.6877730Z source/server/admin/prometheus_stats.cc: In static member function ‘static absl::Status Envoy::Server::PrometheusStatsFormatter::validateParams(const Envoy::Server::StatsParams&)’:
2024-01-09T15:46:32.6880770Z source/server/admin/prometheus_stats.cc:329:1: error: control reaches end of non-void function [-Werror=return-type]
```

So annoying; the code is correct. But as a workaround I suggest making absl::Status a temp and set that temp in the switch statement, fall off the end, and return it.